lib/Service/ParticipantService.php

```diff
@@ -396,6 +397,30 @@ public function addGroup(Room $room, IGroup $group, array $existingParticipants
 		$this->addUsers($room, $newParticipants);
 	}
 
+	/**
+	 * @param string $circleId
+	 * @param string $userId
+	 * @return Circle
+	 * @throws ParticipantNotFoundException
+	 */
+	public function getCircle(string $circleId, string $userId): Circle {
+		try {
+			$circle = Circles::detailsCircle($circleId);
+		} catch (\Exception $e) {
+			throw new ParticipantNotFoundException('Circle not found');
+		}
+
+		// FIXME use \OCA\Circles\Manager::getLink() in the future
+		$membersInCircle = $circle->getInheritedMembers();
+		foreach ($membersInCircle as $member) {
+			if ($member->getUserType() === Member::TYPE_USER && $member->getUserId() === $userId) {
+				return $circle;
```

Comment:
Overkill: check also that `$member->isLocal()`

Reply:
getUserType not covering this?

Reply:
in GlobalScale setup, userId might be identical while having different `getIinstance()`. isLocal() confirm the user is local